pkgs/development/tools/language-servers/nixd/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv
+, meson
+, ninja
+, cmake
+, pkg-config
+, lit
+, nixUnstable
+, boost182
+, gtest
+, llvmPackages_16
+, libbacktrace
+, lib
+, fetchFromGitHub
+}:
+
+let
+  llvmPackages = llvmPackages_16;
+in
+stdenv.mkDerivation rec {
+  pname = "nixd";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "nix-community";
+    repo = pname;
+    rev = version;
+    hash = "sha256-kTDPbsQi9gzFAFkiAPF+V3yI1WBmILEnnsqdgHMqXJA=";
+  };
+
+  mesonBuildType = "release";
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    cmake
+    pkg-config
+
+    # Testing only
+    lit
+
+    nixUnstable.dev
+    boost182.dev
+    gtest.dev
+    llvmPackages.llvm.dev
```

Comment:
The platform of `nix`, `boost` and `llvm` is the target platform instead of the build platform, maybe better to put in `buildInputs`.

Reply:
Do we need these `.dev` outputs because they provide `.pc` files & headers (for meson)?